C7Engine/AI/Pathing/DijkstrasAlgorithm.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections.Generic;
+using System.Linq;
+using C7GameData;
+
+namespace C7Engine.Pathing {
+
+	public class DijkstrasAlgorithm: PathingAlgorithm {
+		private readonly EdgeWalker<Tile> edgeWalker;
+
+		public DijkstrasAlgorithm(EdgeWalker<Tile> edgeWalker) {
+			this.edgeWalker = edgeWalker;
+		}
+
+		public override TilePath PathFrom(Tile start, Tile destination) {
+			Dictionary<Tile, Edge<Tile>> walkingMap = run(start, destination, edgeWalker);
+			return makePath(walkingMap, destination);
+		}
+
+		/**
+		 * return Dictionary: Node -> Edge(previousNode, node, distanceToNodeFromStart)
+		 * when stopWhenReachDestination == false it calculates distances to each available point
+		 */
+		public static Dictionary<TNode, Edge<TNode>> run<TNode>(TNode start, TNode destination,
+			EdgeWalker<TNode> edgeWalker, bool stopWhenReachDestination = false)
```

Comment:
Based on how it's used currently, i.e. with no caching, wouldn't it make more sense to make stopWhenReachDestination default to true?  Is there any benefit to building the whole tree if we don't reuse it?

I suspect rebuilding the whole tree is the reason that even with the binary heap improvement, I'm still seeing the "Explorer pathing took 179 milliseconds" warnings occasionally - it's saving time from the better data structure, but it's calculating the whole tree whereas it wasn't previously.

Reply:
It's a mistake. I should use stopWhenReachDestination = true by default, will fix it.

Reply:
Yup, that makes a huge difference.  With it as true, there are no log statements in 140+ turns that hit the explorer warning time threshold.